Python/compile.c

```diff
@@ -2075,6 +2104,68 @@ wrap_in_stopiteration_handler(struct compiler *c)
     return SUCCESS;
 }
 
+static int
+compiler_type_params(struct compiler *c, asdl_typeparam_seq *typeparams)
+{
+    if (!typeparams) {
+        return SUCCESS;
+    }
+    Py_ssize_t n = asdl_seq_LEN(typeparams);
+
+    for (Py_ssize_t i = 0; i < n; i++) {
+        typeparam_ty typeparam = asdl_seq_GET(typeparams, i);
+        location loc = LOC(typeparam);
+        switch(typeparam->kind) {
+        case TypeVar_kind:
+            ADDOP_LOAD_CONST(c, loc, typeparam->v.TypeVar.name);
+            if (typeparam->v.TypeVar.bound) {
+                expr_ty bound = typeparam->v.TypeVar.bound;
+                if (compiler_enter_scope(c, typeparam->v.TypeVar.name, COMPILER_SCOPE_TYPEPARAMS,
+                                        (void *)typeparam, bound->lineno) == -1) {
+                    return ERROR;
+                }
+                VISIT(c, expr, bound);
```

Comment:
I think this should be `VISIT_IN_SCOPE` so we exit the scope in case of error.
```suggestion
                VISIT_IN_SCOPE(c, expr, bound);
```